MySqlChunkedSnapshotIT - shouldSnapshotMultipleTablesChunkedWithVaryingRowCounts flaky

Failure:
```
2026-03-29T16:00:45.3221347Z [INFO] Results:
2026-03-29T16:00:45.3221660Z [INFO] 
2026-03-29T16:00:45.3221950Z [ERROR] Failures: 
2026-03-29T16:00:45.3223276Z [ERROR]   MySqlChunkedSnapshotIT>AbstractChunkedSnapshotTest.shouldSnapshotMultipleTablesChunkedWithVaryingRowCounts:288->AbstractChunkedSnapshotTest.assertRecordsSnapshotMarkers:696 
2026-03-29T16:00:45.3224185Z expected: "last_in_data_collection"
2026-03-29T16:00:45.3224433Z  but was: "last"
2026-03-29T16:00:45.3224607Z [INFO] 
2026-03-29T16:00:45.3224839Z [ERROR] Tests run: 415, Failures: 1, Errors: 0, Skipped: 66
```

Test Snapshot Output:
```
2026-03-29T15:48:29.1227270Z 2026-03-29 15:48:29,033 INFO   MySQL||0|snapshot|  Snapshot step 7 - Snapshotting data   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1229651Z 2026-03-29 15:48:29,033 INFO   MySQL||0|snapshot|  For table 'chunked_snapshot_test_koigf4.dbz1220a' using select statement: 'SELECT `id`, `data` FROM `chunked_snapshot_test_koigf4`.`dbz1220a`'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1232378Z 2026-03-29 15:48:29,034 INFO   MySQL||0|snapshot|  Estimated row count for table chunked_snapshot_test_koigf4.dbz1220a is OptionalLong[9980]   [io.debezium.connector.binlog.BinlogSnapshotChangeEventSource]
2026-03-29T15:48:29.1235087Z 2026-03-29 15:48:29,034 INFO   MySQL||0|snapshot|  For table 'chunked_snapshot_test_koigf4.dbz1220b' using select statement: 'SELECT `id`, `data` FROM `chunked_snapshot_test_koigf4`.`dbz1220b`'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1237976Z 2026-03-29 15:48:29,035 INFO   MySQL||0|snapshot|  Estimated row count for table chunked_snapshot_test_koigf4.dbz1220b is OptionalLong[9980]   [io.debezium.connector.binlog.BinlogSnapshotChangeEventSource]
2026-03-29T15:48:29.1240816Z 2026-03-29 15:48:29,035 INFO   MySQL||0|snapshot|  For table 'chunked_snapshot_test_koigf4.dbz1220c' using select statement: 'SELECT `id`, `data` FROM `chunked_snapshot_test_koigf4`.`dbz1220c`'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1243533Z 2026-03-29 15:48:29,037 INFO   MySQL||0|snapshot|  Estimated row count for table chunked_snapshot_test_koigf4.dbz1220c is OptionalLong[9980]   [io.debezium.connector.binlog.BinlogSnapshotChangeEventSource]
2026-03-29T15:48:29.1246541Z 2026-03-29 15:48:29,037 INFO   MySQL||0|snapshot|  For table 'chunked_snapshot_test_koigf4.dbz1220d' using select statement: 'SELECT `id`, `data` FROM `chunked_snapshot_test_koigf4`.`dbz1220d`'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1249262Z 2026-03-29 15:48:29,038 INFO   MySQL||0|snapshot|  Estimated row count for table chunked_snapshot_test_koigf4.dbz1220d is OptionalLong[0]   [io.debezium.connector.binlog.BinlogSnapshotChangeEventSource]
2026-03-29T15:48:29.1251438Z 2026-03-29 15:48:29,038 INFO   MySQL||0|snapshot|  Creating snapshot worker pool with 2 worker thread(s)   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1253538Z 2026-03-29 15:48:29,038 INFO   MySQL|||snapshot|  Exporting data from table 'chunked_snapshot_test_koigf4.dbz1220a' (1 of 4 tables)   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.1255976Z 2026-03-29 15:48:29,038 INFO   MySQL|||snapshot|  Exporting data from table 'chunked_snapshot_test_koigf4.dbz1220b' (2 of 4 tables)   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.2041906Z 2026-03-29 15:48:29,139 INFO   MySQL|||snapshot|  	 Finished exporting 10000 records for table 'chunked_snapshot_test_koigf4.dbz1220a' (1 of 4 tables); total duration '00:00:00.101'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.2047598Z 2026-03-29 15:48:29,140 INFO   MySQL|||snapshot|  Exporting data from table 'chunked_snapshot_test_koigf4.dbz1220c' (3 of 4 tables)   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.2050055Z 2026-03-29 15:48:29,149 INFO   MySQL|||snapshot|  	 Finished exporting 10000 records for table 'chunked_snapshot_test_koigf4.dbz1220b' (2 of 4 tables); total duration '00:00:00.111'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.2052683Z 2026-03-29 15:48:29,150 INFO   MySQL|||snapshot|  Exporting data from table 'chunked_snapshot_test_koigf4.dbz1220d' (4 of 4 tables)   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.3043341Z 2026-03-29 15:48:29,245 INFO   MySQL|||snapshot|  	 Finished exporting 10000 records for table 'chunked_snapshot_test_koigf4.dbz1220c' (3 of 4 tables); total duration '00:00:00.105'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.3045319Z Mar 29, 2026 3:48:29 PM com.github.shyiko.mysql.binlog.BinaryLogClient resolveDatabaseVersion
2026-03-29T15:48:29.3046415Z INFO: Database version: 9.1.0 (major=9, minor=1, mariadb=false)
2026-03-29T15:48:29.3047273Z Mar 29, 2026 3:48:29 PM com.github.shyiko.mysql.binlog.BinaryLogClient requestBinaryLogStreamMysql
2026-03-29T15:48:29.3048300Z INFO: Requesting streaming from position filename: mysql-bin.000003, position: 157573114
2026-03-29T15:48:29.3049194Z Mar 29, 2026 3:48:29 PM com.github.shyiko.mysql.binlog.BinaryLogClient connect
2026-03-29T15:48:29.3050030Z INFO: Connected to localhost:3306 at mysql-bin.000003/157573114 (sid:18765, cid:770)
2026-03-29T15:48:29.3053006Z 2026-03-29 15:48:29,255 INFO   MySQL|||snapshot|  	 Finished exporting 10000 records for table 'chunked_snapshot_test_koigf4.dbz1220d' (4 of 4 tables); total duration '00:00:00.105'   [io.debezium.relational.RelationalSnapshotChangeEventSource]
2026-03-29T15:48:29.3055576Z 2026-03-29 15:48:29,255 INFO   MySQL||0|snapshot|  Requested thread factory for component JdbcConnection, id = JdbcConnection named = jdbc-connection-close   [io.debezium.util.Threads]
2026-03-29T15:48:29.3057660Z 2026-03-29 15:48:29,256 INFO   MySQL||0|snapshot|  Creating thread debezium-jdbcconnection-JdbcConnection-jdbc-connection-close   [io.debezium.util.Threads]
2026-03-29T15:48:29.3059091Z 2026-03-29 15:48:29,256 INFO   ||||  Connection gracefully closed   [io.debezium.jdbc.JdbcConnection]
2026-03-29T15:48:29.3060381Z 2026-03-29 15:48:29,257 INFO   MySQL||0|snapshot|  Snapshot - Final stage   [io.debezium.pipeline.source.AbstractSnapshotChangeEventSource]
2026-03-29T15:48:29.3061919Z 2026-03-29 15:48:29,257 INFO   MySQL||0|snapshot|  Snapshot completed   [io.debezium.pipeline.source.AbstractSnapshotChangeEventSource]
...
2026-03-29T15:48:32.4838684Z [ERROR] io.debezium.connector.mysql.MySqlChunkedSnapshotIT.shouldSnapshotMultipleTablesChunkedWithVaryingRowCounts -- Time elapsed: 10.73 s <<< FAILURE!
2026-03-29T15:48:32.4840275Z org.opentest4j.AssertionFailedError: 
2026-03-29T15:48:32.4840619Z 
2026-03-29T15:48:32.4840774Z expected: "last_in_data_collection"
2026-03-29T15:48:32.4841181Z  but was: "last"
2026-03-29T15:48:32.4842130Z 	at io.debezium.pipeline.AbstractChunkedSnapshotTest.assertRecordsSnapshotMarkers(AbstractChunkedSnapshotTest.java:696)
2026-03-29T15:48:32.4844144Z 	at io.debezium.pipeline.AbstractChunkedSnapshotTest.shouldSnapshotMultipleTablesChunkedWithVaryingRowCounts(AbstractChunkedSnapshotTest.java:288)
2026-03-29T15:48:32.4845893Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2026-03-29T15:48:32.4846612Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
2026-03-29T15:48:32.4847444Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

From https://productionresultssa13.blob.core.windows.net/actions-results/caa4732e-c066-4539-ac03-e3613067aa0a/workflow-job-run-7752efbb-c44e-5fbe-9c54-11c34454abe3/logs/job/job-logs.txt?rsct=text%2Fplain&se=2026-03-30T05%3A09%3A49Z&sig=MDYLcA0E33BRd0sZaCj6whZNEaz86oHgtxkkdlzUSsE%3D&ske=2026-03-30T07%3A05%3A43Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2026-03-30T03%3A05%3A43Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-11-05&sp=r&spr=https&sr=b&st=2026-03-30T04%3A59%3A44Z&sv=2025-11-05

--- a/debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogChunkedSnapshotIT.java
+++ b/debezium-connector-binlog/src/test/java/io/debezium/connector/binlog/BinlogChunkedSnapshotIT.java
@@ -45,15 +45,7 @@ public abstract class BinlogChunkedSnapshotIT<T extends SourceConnector>
 
         connection = getTestDatabaseConnection(DATABASE.getDatabaseName());
         if (connection.connection().getAutoCommit()) {
-            // todo:
-            // for some reason enabling auto-commit here creates issues for other test classes
-            // that come after this test class; despite the fact of whether the buffer size is
-            // set to 0 to disable it or if we use auto-commit within try-with-resources areas
-            // that perform bulk database operations. For now, disabling this, as the only
-            // impact is that loading data in the tests takes significantly longer.
-            // Makes sure that when we do large bulk inserts, the performance is optimal
-            // and the inserts are all part of a singular transaction.
-            // connection.setAutoCommit(false);
+            connection.setAutoCommit(false);
         }
 
         super.beforeEach();

